Pearson-Advance/frontend-app-lti
# feat: add table to list course ccx

> [!CAUTION]
> To avoid a very long PR I split the effort of solving the ticket PAD-1543 into three. **This is the third one** 
> The PRs are accumulative and need the changes from the previous PR; in their `base` branch they will be pointing to the branch of the previous PR so they show only the changes applied in that specific PR

> [!NOTE]
> Once the previous PR is merged, the base of this branch will change to `main`

This PR adds a table to list the classes of the clicked institution-licensed course

### Main changes

- [src/features/Courses/hooks/useManageTableSelection.js](https://github.com/Pearson-Advance/frontend-app-lti/pull/4/files#diff-9fc730d72d28dcdc06d1ecd7abff0b83478723378aefb4ec05d2a89a1035b6c4): Abstracts common logic about courses selection and the deep linking validation
- [src/features/Courses/hooks/useSearchBar.js](https://github.com/Pearson-Advance/frontend-app-lti/pull/4/files#diff-4be26a3935ffe2acfe54141d22c17451b9f13330691bcd7d809708eb6af84a97): Abstracts common logic about search bar use and pagination
- [src/features/Courses/TableLayout/index.jsx](https://github.com/Pearson-Advance/frontend-app-lti/pull/4/files#diff-112ab8190034c80141c9f04c0ca42693b9974d81916932313ced81e0df7c163e): Reusable component for creating tables 

### How to test
- Set the environment to use `Deep Linking`
- Enable CCX in a course and create a Class of that course
- Enable the redirect to the LTI MFE when launching a `Deep Linking Flow`
- Up the LTI MFE 

If everything goes well, once you launch a Deep Linking Flow (I use IMS), you should be redirected to the LTI MFE and watch the courses table with the master courses you created before. Now if you click the `View class list` button, you should watch a table with the list of classes (CCX) from this specific course

![image](https://github.com/user-attachments/assets/3e5dbe28-d0b2-469c-ac61-23b7270a3346)


### JIRA Issue
PAD-1543
